core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -985,54 +955,43 @@ private MutationSet sendMutationsToTabletServer(String location,
         throw new IOException(e);
       }
     }
+
+    private void close() {
+      this.closed = true;
+      binningThreadPool.shutdownNow();
+      sendThreadPool.shutdownNow();
+    }
   }
 
   // END code for sending mutations to tablet servers using background threads
 
-  private static class MutationSet {
+  private static class MutationSet extends ConcurrentHashMap<TableId,Queue<Mutation>> {
```

Comment:
These concurrent data structs are great at increasing the throughput of highly concurrent applications, however they are slower for single threaded uses cases because of heavy use of volatile pointers.  There are concurrent and single threaded uses of this class, so not sure if this changes helps, hurts, or is neutral.

If the type of incomingQueue were changed to BlockingQueue as suggested in another comment that may remove the concurrent use of this type, but not sure.

Reply:
I changed `incomingQueue` to a BlockingQueue and `MutationSet` to extend HashMap.